pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -114,72 +115,69 @@ func (r *ReconcileNSTemplateTier) Reconcile(request reconcile.Request) (reconcil
+			for _, account := range mur.Spec.UserAccounts {
+				accountTmpls := account.Spec.NSTemplateSet
+				logger.Info("checking MasterUserRecord account", "name", mur.Name, "tier", accountTmpls.TierName)
+				if accountTmpls.TierName != instance.Name {
+					// skip if the user account is not based on the current Tier
 					continue
-				} else if !errors.IsNotFound(err) {
-					logger.Error(err, "unable to get TemplateUpdateRequest for MasterUserRecord", "name", mur.Name)
-					return reconcile.Result{}, err
 				}
-				for _, account := range mur.Spec.UserAccounts {
-					accountTmpls := account.Spec.NSTemplateSet
-					logger.Info("checking MasterUserRecord account", "name", mur.Name, "tier", accountTmpls.TierName)
-					if accountTmpls.TierName != instance.Name {
-						// skip if the user account is not based on the current Tier
-						continue
-					}
-					// compare all TemplateRefs in the current MUR.UserAccount vs NSTemplateTier instance
-					if !reflect.DeepEqual(templateRefs(accountTmpls), templateRefs(instance.Spec)) {
-						logger.Info("creating a TemplateUpdateRequest to update the MasterUserRecord", "name", mur.Name, "tier", instance.Name)
-						if err = r.client.Create(context.TODO(), &toolchainv1alpha1.TemplateUpdateRequest{
-							ObjectMeta: metav1.ObjectMeta{
-								Namespace: instance.Namespace,
-								Name:      mur.Name,
-								Labels: map[string]string{
-									toolchainv1alpha1.NSTemplateTierNameLabelKey: instance.Name,
-								},
+				// compare all TemplateRefs in the current MUR.UserAccount vs NSTemplateTier instance
+				if !reflect.DeepEqual(templateRefsFor(accountTmpls), templateRefsFor(instance.Spec)) {
+					logger.Info("creating a TemplateUpdateRequest to update the MasterUserRecord", "name", mur.Name, "tier", instance.Name)
+					if err = r.client.Create(context.TODO(), &toolchainv1alpha1.TemplateUpdateRequest{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: instance.Namespace,
+							Name:      mur.Name,
+							Labels: map[string]string{
+								toolchainv1alpha1.NSTemplateTierNameLabelKey: instance.Name,
 							},
-							Spec: toolchainv1alpha1.TemplateUpdateRequestSpec{
-								TierName:         instance.Name,
-								Namespaces:       instance.Spec.Namespaces,
-								ClusterResources: instance.Spec.ClusterResources,
-							},
-						}); err != nil {
-							return reconcile.Result{}, err
-						}
-
-						// the controller creates a single TemplateUpdateRequest resource per reconcile loop,
-						// so the request has to be requeued, in order to create other TemplateUpdateRequests
-						return reconcile.Result{Requeue: true}, nil
+						},
+						Spec: toolchainv1alpha1.TemplateUpdateRequestSpec{
+							TierName:         instance.Name,
+							Namespaces:       instance.Spec.Namespaces,
+							ClusterResources: instance.Spec.ClusterResources,
+						},
+					}); err != nil {
+						return reconcile.Result{}, err
```

Comment:
And here too.
But you know what? Maybe it's worth moving all this code to a method called `ensureTemplateUpdateRequests()` and if it returns an error then wrap it in the status.